### PR TITLE
refactor(list-collections): support retryable reads

### PR DESCRIPTION
**Pull request by mbroadst**
## Description
Add retryable reads support to `listColections`

**What changed?**
- Added `ListCollectionsOperation`

### CODE REVIEW

File: lib/operations/list_collections.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const maxWireVersion = require('../core/utils').maxWireVersion;
+const listCollectionsTransforms = require('./db_ops').listCollectionsTransforms;
+const CONSTANTS = require('../constants');
+
+const LIST_COLLECTIONS_WIRE_VERSION = 3;
+
+class ListCollectionsOperation extends CommandOperationV2 {
+  constructor(db, filter, options) {
+    super(db, options);
+    this.options.full = true;
```
Comment (daprahamian) [inline lib/operations/list_collections.js]: why are we doing this?

Comment (mbroadst) [reply lib/operations/list_collections.js]: It's being used [here](https://github.com/mongodb/node-mongodb-native/blob/next/lib/operations/command_v2.js#L38) to inform the base class whether it should return a complete server response, or the `result` field of the response. All methods that are used with cursors require a full response. 

This actually doesn't need to be communicated through `options`, I'll change all of these to not use `options` in a follow-up PR.

File: lib/operations/list_collections.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        filter = { $and: [{ name: filter.name }, { name: /^((?!\$).)*$/ }] };
+      } else {
+        filter = { name: /^((?!\$).)*$/ };
+      }
+
+      const transforms = listCollectionsTransforms(databaseName);
+      server.query(
+        `${databaseName}.${CONSTANTS.SYSTEM_NAMESPACE_COLLECTION}`,
+        { query: filter },
+        { batchSize: this.batchSize || 1000 },
```
Comment (kvwalker) [inline lib/operations/list_collections.js]: Where is this default coming from?

File: lib/operations/list_collections.js (hunk 1 above)
Comment (mbroadst) [reply lib/operations/list_collections.js]: https://jira.mongodb.org/browse/NODE-1744

File: lib/operations/list_collections.js (hunk 2 above)
Comment (mbroadst) [reply lib/operations/list_collections.js]: This is the default batch size for find operations. This whole section of code will go away once I can reuse a `FindOperation`, so I wouldn't worry _too_ much about it yet.

**Pull request closed without merging**